libs/components/src/lib/indoor-maps.component.ts

```diff
@@ -0,0 +1,401 @@
+            zoom: 21,
+            maxZoom: 26,
+        };
+        this.map_view_instance = new mapsindoors.mapView.GoogleMapsView(
+            this.map_view_options
+        );
+
+        this.mapsIndoors_instance = new mapsindoors.MapsIndoors({
+            mapView: this.map_view_instance,
+        });
+        return (this.googleMaps_instance = this.map_view_instance.getMap());
+    }
+
+    initDirections() {
+        const externalDirectionsProvider =
+            new mapsindoors.directions.GoogleMapsProvider();
+        this.mapsIndoors_directions_service_instance =
+            new mapsindoors.services.DirectionsService(
+                externalDirectionsProvider
+            );
+        const directionsRendererOptions = {
+            mapsIndoors: this.mapsIndoors_instance,
+        };
+        this.mapsIndoors_directions_renderer_instance =
+            new mapsindoors.directions.DirectionsRenderer(
+                directionsRendererOptions
+            );
+    }
+
+    selectFloors() {
+        const floorSelectorElement = document.createElement('div');
+        new mapsindoors.FloorSelector(
+            floorSelectorElement,
+            this.mapsIndoors_instance
+        );
+        this.googleMaps_instance.controls[
+            google.maps.ControlPosition.RIGHT_TOP
+        ].push(floorSelectorElement);
+    }
+
+    async onSearch(): Promise<any> {
+        const searchParams = { q: this.searchElement.nativeElement.value };
+        await mapsindoors?.services.LocationsService.getLocations(
+            searchParams
+        ).then((locations: any[]) => {
+            this.search_result_items = locations;
+        });
+    }
+
+    private _getUserLocation() {
+        const options = { timeout: 10000, enableHighAccuracy: true };
+
+        return new Promise<GeolocationPosition>(async (resolve, reject) => {
+            if ('geolocation' in navigator) {
+                await navigator.geolocation.getCurrentPosition(
+                    (position: GeolocationPosition) => {
+                        this.user_latitude = position.coords.latitude;
+                        this.user_longitude = position.coords.longitude;
+                        console.log(
+                            this.user_latitude,
+                            this.user_longitude,
+                            'user geolocation'
+                        );
+                        resolve(position);
+                    },
+                    (error) => {
+                        this.geolocation_error_message =
+                            'Error: ' +
+                            error.message?.toString() +
+                            '. Please enable geolocation settings.';
+                        reject(error);
+                    },
+                    options
+                );
+                navigator.geolocation.watchPosition(
+                    this._updateGeolocation.bind(this),
```

Comment:
It's slightly cleaner to use anonymous function here instead of bind.
`(_) => this._updateGeolocation(_)`

Reply:
I've updated this now.